lib/schema-registry.js

```diff
@@ -84,6 +84,15 @@ var SchemaRegistry = module.exports = cip.extend(function(opts) {
   this.schemaTypeById = {};
 });
 
+function typeFromSchemaResponse(schema, parseOptions) {
+  var schemaType = avro.parse(schema);
+  if(parseOptions.registry && parseOptions.registry[schemaType.name] != undefined){
+    return parseOptions.registry[schemaType.name];
+  } else {
+    return avro.parse(schema, parseOptions);
+  }
+}
```

Comment:
Hmmm.... It's better a more defensive type check here to avoid coertion problems. What about ->
```
if(typeof parseOptions.registry === 'object' && typeof parseOptions.registry[schemaType.name] !== 'undefined'){
    return parseOptions.registry[schemaType.name];
}
return avro.parse(schema, parseOptions);
```